aerospike/datadog_checks/aerospike/aerospike.py

```diff
@@ -274,10 +274,16 @@ def collect_latency(self, namespaces):
                 continue
 
             # match only works at the beginning
-            ns_metric_name_match = re.match(r'{(\w+)}-(\w+):', line)
+            # ':' or ';' are not allowed in namespace-name: https://www.aerospike.com/docs/guide/limitations.html
+            ns_metric_name_match = re.match(r'{([^\}:;]+)}-(\w+):', line)
```

Comment:
Awesome. Could you take the opportunity to compile the regex in init?

Reply:
@therve Regexes are cached by python, not sure it worth moving them away from where they are used.

Reply:
Not really in a reliable fashion though. If you have another integration in the same agent uses lots of regexp, then your cache is gone.

I think moving them a bit but giving them a readable name makes sense.

Reply:
```§ python2 -c 'import re; print(re._MAXCACHE)'
100
§ python3 -c 'import re; print(re._MAXCACHE)'
512```
The regex cache is quite small. Yeah, probably better to not rely on the cache.

If you don't mind, I will do a refactoring to pre compile all the regexes in a separate PR. We might want to cherry pick this PR (probably not, just in case).